### DISCUSSION TITLE
Big difference in pv.Plotter() vs PolyData.plot images - how to fix

### DISCUSSION THREAD

**Discussion by slartz42** [pyvista/pyvista#3772] [Q&A] [answered]
I'm still fairly new to pyvista and Python in general, I apologize if this question has been asked already but I can't find any similar questions.

I am currently using a PolyData.extrude_rotate function to convert a 2D polygon to 3D and the results look excellent when I use the PolyData.plot function.

When I use pv.Plotter() with add_mesh however, the image does not look the same. No matter what I try, I cannot get the edges to show and the results are much worse:

![PolyData_plot](https://user-images.githubusercontent.com/52103361/210269370-b9715ae1-db81-419d-bf39-c949839cb62b.PNG)
![pv_Plotter](https://user-images.githubusercontent.com/52103361/210269374-50485daf-276f-4f9b-b98d-5ecee8bf91bf.PNG)

Here is what I have tried without success:
- extract_surface
- show_edges
- clean

 I need to use pv.Plotter() so I can plot multiple objects at the same time, but I'm not sure how to resolve this issue. Is there a way to plot multiple objects with PolyData.plot that I'm unaware of? 

Any help would be greatly appreciate

**Comment by banesullivan** [comment, marked as answer]
Can you share the exact code your using and perhaps the mesh file as well?

**Reply by slartz42** [reply]
Sure, here the code for the PolyData.plot is below. I'm getting a list of xy coordinates from a shapely polygon so I don't really have a mesh file. I'm appending an array of zeros to the shapely coordinates so the extrude_rotate function has Z coordinates to use, the code below generates the first picture in my previous post

```
workpiece_3d = segmentize(workpiece_polygon, 0.1)
workpiece_exterior = np.array((list(workpiece_3d.exterior.coords)))
empty_z = np.zeros((workpiece_exterior.shape[0], 1))
workpiece_exterior = np.concatenate((workpiece_exterior, empty_z), axis=1)
workpiece_profile = pv.PolyData(workpiece_exterior)
workpiece_extruded = workpiece_profile.extrude_rotate(resolution=500, angle=360, capping=True, rotation_axis=(1, 0, 0))
workpiece_extruded.plot(eye_dome_lighting=True, color='silver', smooth_shading='True')
```

Here is the code that generates the model with no edges:

```
pl = pv.Plotter()
pl.add_mesh(workpiece_extruded, color='silver')
pl.show()
```

**Reply by banesullivan** [reply]
The difference is in the `eye_dome_lighting=True` and `smooth_shading=True` options. The snippet with the `Plotter` is not using those. 

Try:

```py
pl = pv.Plotter()
pl.add_mesh(workpiece_extruded, color='silver', smooth_shading=True)
pl.enable_eye_dome_lighting()
pl.show()
```

And you should get the same result.

**Reply by slartz42** [reply]
That works, can't believe I missed that. Thanks!

**Reply by banesullivan** [reply]
All good. I also want to add a note on why the initial plot you had looked so... not great...

The source mesh is a point cloud, and the resulting extruded mesh is also a point cloud. though very dense (as far as I can tell from looking at this code). Since it is a dense point cloud, the default lighting looks bad and requires special rendering techniques like eye dome lighting. 

This example has a nice overview of the phenomena: https://docs.pyvista.org/examples/02-plot/edl.html

**Reply by slartz42** [reply]
Sorry to bring up an old thread, but what would be the easiest way to convert this point cloud to a mesh? I've been trying for a while and struggling to find a solution

**Reply by banesullivan** [reply]
Would you please open a new discussion topic for that and please share the point cloud file (can be done with `workpiece_extruded .save('workpiece_extruded.vtk')`)
